slobrok/src/vespa/slobrok/server/map_view.h

```diff
@@ -0,0 +1,51 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "service_mapping.h"
+#include "map_diff.h"
+#include "map_source.h"
+#include <vespa/vespalib/util/gencnt.h>
+#include <map>
+#include <mutex>
+#include <set>
+
+namespace slobrok {
+
+/**
+ * @class MapView
+ * @brief Holds a name->spec map which can be incrementally updated
+ **/
+class MapView : public MapSource
```

Comment:
MapView sounds a bit generic. Other options could be RemoteMapSource (we do it for the events) or ServiceMapMirror or similar (to match it with its partner; ServiceMapHistory)